File: examserver/src/main/java/com/examportal/service/QuizService.java
package com.examportal.service;

import java.util.List;
import java.util.Set;

import com.examportal.model.quiz.Category;
import com.examportal.model.quiz.Quiz;

public interface QuizService {

	public Quiz addQuiz(Quiz quiz);

	public Quiz updateQuiz(Quiz quiz);

	public Set<Quiz> getAllQuiz();

	public Quiz getQuiz(Long quizId);

	public void deleteQuiz(Long quizId);

	public List<Quiz> getQuizzesOfCategory(Category category);

	public List<Quiz> getActiveQuizzes();

	public List<Quiz> getActiveQuizzesOfCategory(Category category);
}

File: examfront/src/app/services/quiz/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from '../helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _http: HttpClient) { }

  public quizzes() {
    return this._http.get(`${baseUrl}/quiz/`);
  }

  public addQuiz(quiz) {
    return this._http.post(`${baseUrl}/quiz/`, quiz);
  }

  public deleteQuiz(qId) {
    return this._http.delete(`${baseUrl}/quiz/${qId}`);
  }

  public getQuiz(qId) {
    return this._http.get(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quiz) {
    return this._http.put(`${baseUrl}/quiz/`, quiz);
  }

  public getQuizofCategory(cId) {
    return this._http.get(`${baseUrl}/quiz/category/${cId}`);
  }

  public getActiveQuizzes() {
    return this._http.get(`${baseUrl}/quiz/active`);
  }

  public getActiveQuizzesofCategory(cId) {
    return this._http.get(`${baseUrl}/quiz/category/active/${cId}`);
  }
}